cmd/api/database_test.go

```diff
@@ -97,40 +97,71 @@ func TestListBooks(t *testing.T) {
 		teardownDB(t)
 	})
 
+	var testBookslist []Book
+	listSize := 11
+
 	t.Run("List books without errors even if there is no books in the database", func(t *testing.T) {
 		is := is.New(t)
 
 		// Write the List Books test here.
-		returnedBooks, err := listBooks()
+		returnedBooks, err := listBooks("", 0.00, 9999.99)
 		is.NoErr(err)
 		is.Equal(returnedBooks, []Book{})
 	})
 
-	t.Run("List books without errors", func(t *testing.T) {
+	t.Run("List all books, no filtering, without errors", func(t *testing.T) {
 		is := is.New(t)
 
 		// Setting up, creating books to be listed.
-		var bookslist []Book
-		listSize := 11
 
 		for i := 0; i < listSize; i++ {
 			b := Book{
 				ID:        uuid.New(),
 				Name:      fmt.Sprintf("Book number %06v", i),
-				Price:     toPointer(float32(40.0)),
-				Inventory: toPointer(10),
+				Price:     toPointer(float32((i * 100) + 1)),
+				Inventory: toPointer(i + 1),
 			}
 
 			newBook, err := storeOnDB(b)
 			is.NoErr(err)
 			is.Equal(newBook, b)
-			bookslist = append(bookslist, b)
+			testBookslist = append(testBookslist, b)
 		}
```

Comment:
All the following tests depend on this `for` loop that creates the books to be listed.
This means that the other tests depend on this test to run correctly, lets move this `for` loop to line `111`, that way you set up all the tests in one spot, outside of the `t.Run` function.